security/sanitizers.rst

```diff
@@ -0,0 +1,442 @@
+pick up the flags defined in the ``[conf]`` section and apply them to the build.
+
+Managing sanitizers with a CMake toolchain
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Besides using Conan profiles to manage sanitizer settings, you can also use other approaches.
+
+If you already have a :ref:`custom CMake toolchain file <conan_cmake_user_toolchain>` to manage compiler
+and build options, you can pass the necessary flags to enable sanitizers there instead of profiles.
+
+.. code-block:: cmake
+   :caption: cmake/my_toolchain.cmake
+
+   # Apply to all targets; consider per-target options for finer control
+   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address,undefined -fno-omit-frame-pointer")
+   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address,undefined -fno-omit-frame-pointer")
+   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address,undefined")
+   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address,undefined")
+
+Then, specify this toolchain file as part of your Conan profile:
+
+.. code-block:: ini
+   :caption: profiles/asan_ubsan
+
+   include(default)
+
+   [settings]
+   build_type=Debug
+   compiler.sanitizer=AddressUndefinedBehavior
+
+   [conf]
+   tools.cmake.cmaketoolchain:user_toolchain=cmake/my_toolchain.cmake
+
+This way, you can keep your existing CMake toolchain file and still leverage Conan profiles to manage other settings.
+
+Building Examples Using Sanitizers
+----------------------------------
+
+To better illustrate this, first, please clone the sources to recreate this project. You can find them in the
+`examples2 repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: bash
+
+   git clone https://github.com/conan-io/examples2.git
+   cd examples2/examples/security/sanitizers/compiler_sanitizers
+
+In this example we will see how to prepare Conan to use sanitizers in different ways.
+
+To show how to use sanitizers in your builds, let's consider two examples.
+
+AddressSanitizer: index out of bounds
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this example, we will build a simple C++ program that intentionally accesses an out-of-bounds index
+in an array, which should trigger ASan when running the program.
+
+.. code-block:: cpp
+   :caption: index_out_of_bounds/main.cpp
+   :emphasize-lines: 11
+
+   #include <iostream>
+   #include <cstdlib>
+
+   int main() {
+   #ifdef __SANITIZE_ADDRESS__
+     std::cout << "Address sanitizer enabled\n";
+   #else
+     std::cout << "Address sanitizer not enabled\n";
+   #endif
+
+     int foo[100];
+     foo[100] = 42; // Out-of-bounds write
+
+     return EXIT_SUCCESS;
+   }
+
+**Note:** The preprocessor check above is portable for GCC, Clang and MSVC.
+The define ``__SANITIZE_ADDRESS__`` is present when **ASan** is active;
+
+**To build and run this example using Conan:**
+
+.. code-block:: bash
+
+   conan export index_out_of_bounds/
```

Comment:
The ``conan export`` is necessary? I think it is not.

Reply:
Simplified by commit 5786822ac616f6c1069577dbe01b3fd9874d38d9